### PR TITLE
Documentation update about Buffer initialization

### LINKED ISSUE #7230: new Buffer() can contain any content
When creating a new Buffer using "new Buffer(size)", the buffer can contain any data. I was assuming the buffer to contain just zeroes, but this assumption is not correct.

If this is behaviour is intended, e.g. because of performance concerns, please update the documentation to reflect the fact that the returned buffer may contain random values.

### PR DESCRIPTION
**Pull request by sarathms**
fixes #7230

### CODE REVIEW

File: doc/api/buffer.markdown [hunk 1]
```diff
@@ -64,7 +64,7 @@ It can be constructed in a variety of ways.
 
 Allocates a new buffer of `size` octets. Note, `size` must be no more than
 [kMaxLength](smalloc.html#smalloc_smalloc_kmaxlength). Otherwise, a `RangeError`
-will be thrown here.
+will be thrown here. Unlike `ArrayBuffers`, the underlying memory for buffers is not initialized. So the contents of a newly created `Buffer` is unknown. Use `buf.fill(0)`to initialize a buffer to zeroes.
```
Comment (cjihrig) [inline doc/api/buffer.markdown]: I would just say:

The underlying memory of a `Buffer` is not initialized. If this behavior is desired, use `Buffer.prototype.fill()`.

Comment (sarathms) [inline doc/api/buffer.markdown]: I added the 'Unlike ArrayBuffers... ' part as suggested by another core dev. I think it helps. If its the length, I can paraphrase it further. But the `Buffer.protoype.fill()` usage is inconsistent with rest of the documentation. Its gonna take a while to change all references to match it. What do you advice?

Comment (cjihrig) [inline doc/api/buffer.markdown]: I don't see a real need to mention `ArrayBuffer`s at all, but I'm not strongly opposed to it. I guess it makes more sense to stick with the existing style of the docs.

Comment (sarathms) [inline doc/api/buffer.markdown]: How about this?
>Unlike, the underlying memory of a Buffer is not initialized. Use 
`buf.fill(0)` to initialize a buffer to zeroes.
There's a comparison with ArrayBuffer in the explanation for `#slice` too. I'd say let's keep it.

Comment (cjihrig) [inline doc/api/buffer.markdown]: @tjfontaine thoughts?

Comment (trevnorris) [inline doc/api/buffer.markdown]: Please wrap at 80 chars.

**Comment by trevnorris** [comment]
One style nit, but LGTM otherwise.

**Comment by jasnell** [comment]
LGTM

**Comment by jasnell** [comment]
Landed in 70efdf3

**Pull request closed without merging**